lib/elixir/lib/file/stream.ex

```diff
@@ -115,8 +116,14 @@ defmodule File.Stream do
       {:error, __MODULE__}
     end
 
+    defp strip_bom(device) do
+      header = IO.binread(device, 4)
+      {:ok, _new_pos} = :file.position(device, File.bom_length(header))
+      device
+    end
+
     defp read_modes(modes) do
-      for mode <- modes, mode not in [:write, :append], do: mode
+      for mode <- modes, mode not in [:write, :append, :strip_bom], do: mode
```

Comment:
Since we're touching this, would `Enum.filter/2` be a better fit here? 🤔 

```elixir
Enum.filter(modes, &(&1 not in [:write, :append, :strip_bom]))
```